server/test/services/withings/lib/withings.handler.test.js

```diff
@@ -0,0 +1,462 @@
+    await assert.isNotNull(resultInit);
+
+    const deviceToPoll = resultInit.next();
+    deviceToPoll.features = [
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.WEIGHT,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.UNKNOWN,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.TEMPERATURE,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.SYSTOLIC_BLOOD_PRESSURE,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.SPO2,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.SKIN_TEMPERATURE,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.PULSE_WAVE_VELOCITY,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.MUSCLE_MASS,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.HYDRATION,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.HEIGHT,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.HEARTH_PULSE,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.FAT_RATIO,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.FAT_MASS_WEIGHT,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.FAT_FREE_MASS,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.DIASTOLIC_BLOOD_PRESSURE,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.BONE_MASS,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.HEALTH.BODY_TEMPERATURE,
+        last_value_changed: new Date(),
+      },
+      {
+        id: uuid.v4(),
+        type: DEVICE_FEATURE_TYPES.SENSOR.INTEGER,
+        last_value_changed: new Date(),
+      },
+    ];
+    deviceToPoll.params = [
+      {
+        device_id: deviceToPoll.id,
+        name: 'withingsDeviceId',
+        value: 'withingsDevideId',
+      },
+    ];
+
+    let resultPoll = await withingsHandler.poll(deviceToPoll);
+    await assert.isNotNull(resultPoll);
```

Comment:
The poll function is no more async, so no need to await.

The test is green even if the function is empty...
![image](https://user-images.githubusercontent.com/32032645/105102774-9f1a2700-5aaf-11eb-8386-dec48a9f2d34.png)

Reply:
In this case I use Promise.map in poll function and I don't know how I can wait result of Promise in test to tests changes

Reply:
Maybe you can check that you call the function "gladys.device.saveHistoricalState"?
If the function changes, the test must be in error.

Reply:
I try lot of thing and finaly resolution of saveHistoricalState arrive always after the end of test execution, I don't understand why ... and how I can fix it